File: backend/app/models/analysis.py
from pydantic import BaseModel


class ConflictNode(BaseModel):
    id: str
    label: str
    description: str
    type: str
    color: Optional[str] = None


class ConflictEdge(BaseModel):
    source_id: str
    target_id: str
    label: str


class ConflictAnalysis(BaseModel):
    id: str
    task_id: str
    center_node: ConflictNode
    satellite_nodes: list[ConflictNode]
    edges: list[ConflictEdge]
    triz_principles: list[str]

File: frontend/src/types/analysis.ts
export interface ConflictNode {
  id: string;
  label: string;
  description: string;
  type: 'center' | 'satellite';
  color?: string;
  sublabel?: string;
  position?: 'top' | 'right' | 'bottom' | 'left';
}

export interface ConflictEdge {
  sourceId: string;
  targetId: string;
  label: string;
}

export interface ConflictAnalysis {
  id: string;
  taskId: string;
  centerNode: ConflictNode;
  satelliteNodes: ConflictNode[];
  edges: ConflictEdge[];
  principles: string[];
}
